File: app/src/main/java/com/bkjcb/rqapplication/EmergencyActivity.java
package com.bkjcb.rqapplication;

import android.content.Context;
import android.content.Intent;

import com.bkjcb.rqapplication.adapter.EmergencyItemAdapter;
import com.bkjcb.rqapplication.model.EmergencyItem;
import com.bkjcb.rqapplication.model.EmergencyItem_;

import java.util.List;

/**
 * Created by DengShuai on 2020/3/5.
 * Description :
 */
public class EmergencyActivity extends ActionRegisterActivity {
    private EmergencyItemAdapter adapter;

    @Override
    protected int setLayoutID() {
        return R.layout.activity_main_check;
    }

    @Override
    protected void createAdapter() {
        adapter = new EmergencyItemAdapter(R.layout.item_checkadapter_view);
        mRegisterList.setAdapter(adapter);
        adapter.bindToRecyclerView(mRegisterList);
        adapter.setOnItemClickListener(this);
    }

    @Override
    protected String getTitleString() {
        return "事故现场";
    }

    protected static void ToActivity(Context context) {
        Intent intent = new Intent(context, EmergencyActivity.class);
        context.startActivity(intent);
    }

    private List<EmergencyItem> queryLocalData() {
        if (isShowAll) {
            return EmergencyItem.getBox().getAll();
        } else {
            return EmergencyItem.getBox().query().notEqual(EmergencyItem_.status, 2).build().find();
        }
    }


    protected void showCheckList() {
        List<EmergencyItem> list = queryLocalData();
        //Toast.makeText(this, list.size()+"", Toast.LENGTH_SHORT).show();
        if (list.size() > 0) {
            adapter.setNewData(list);
        } else {
            adapter.setNewData(null);
            adapter.setEmptyView(createEmptyView());
        }
    }

    @Override
    protected void createNew(int position) {
        if (position >= 0) {
            CreateEmergencyActivity.ToActivity(EmergencyActivity.this, (EmergencyItem) adapter.getItem(position));
        } else {
            CreateEmergencyActivity.ToActivity(EmergencyActivity.this, null);
        }
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/emergency/EmergencyActivity.kt
package com.bkjcb.rqapplication.emergency

import android.content.Context
import android.content.Intent
import android.support.v7.widget.LinearLayoutManager
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.actionregister.ActionRegisterActivity
import com.bkjcb.rqapplication.actionregister.adapter.ActionRegisterItemAdapter
import com.bkjcb.rqapplication.emergency.adapter.EmergencyItemAdapter
import com.bkjcb.rqapplication.emergency.model.EmergencyItem
import com.bkjcb.rqapplication.emergency.model.EmergencyItem_
import kotlinx.android.synthetic.main.activity_main_check.*

/**
 * Created by DengShuai on 2020/6/24.
 * Description :
 */
class EmergencyActivity : ActionRegisterActivity() {
    private var adapter: EmergencyItemAdapter? = null

    companion object {
        fun toActivity(context: Context) {
            val intent = Intent(context, EmergencyActivity::class.java);
            context.startActivity(intent);
        }
    }

    override fun createAdapter() {
        adapter = EmergencyItemAdapter(R.layout.item_checkadapter_view);
        check_list.adapter = adapter;
        check_list.layoutManager = LinearLayoutManager(this)
        adapter?.onItemClickListener = this
        adapter?.bindToRecyclerView(check_list)
    }

    override fun createNew(position: Int) {
        if (position >= 0) {
            CreateEmergencyActivity.ToActivity(this@EmergencyActivity, adapter?.getItem(position))
        } else {
            CreateEmergencyActivity.ToActivity(this@EmergencyActivity, null)
        }
    }

    override fun getTitleString(): String {
        return "事故现场"
    }

    private fun queryLocalData(): List<EmergencyItem?>? {
        return if (isShowAll) {
            EmergencyItem.getBox().all
        } else {
            EmergencyItem.getBox().query().notEqual(EmergencyItem_.status, 2).build().find()
        }
    }

    override fun showCheckList() {
        val list = queryLocalData()
        if (list != null && list.isNotEmpty()) {
            adapter?.setNewData(list)
        } else {
            adapter?.setNewData(null)
            adapter?.emptyView = createEmptyView()
        }
    }
}